scripts/setup_mongo_dev

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```

Comment:
Non-blocking: This script isn't going to have the intended outcome of being runnable and initializing the environment.  It will set the variables, but only in the context of the script runtime.  In order to this to work as such, you would need to write the variables out to an `.env` file, and then have the user source that file in their shell.  That way, everything will be set accordingly.  We do something similar in [rails_local_setup.rb](https://github.com/broadinstitute/single_cell_portal_core/blob/development/rails_local_setup.rb) (it is in Ruby, however the the paradigm holds).  Also, any shell scripts should be suffixed with `.sh` for clarity.

You could do it like this (I've added in some error handling as well to make it more robust):

```suggestion
#! /bin/bash

VAULT_TOKEN_PATH="$1"
if [[ -z "$VAULT_TOKEN_PATH" ]]
then
  echo "You must provide a path to a Github token to proceed"
  exit 1
fi
vault login -method=github token=$(cat $VAULT_TOKEN_PATH)
if [[ $? -ne 0 ]]
then
  echo "Unable to authenticate into vault"
  exit 1
fi

ENV_FILENAME='ingest_creds.env'
export BROAD_USER=`whoami`
echo "export MONGODB_USERNAME='single_cell'" |> $ENV_FILENAME
echo "export DATABASE_NAME='single_cell_portal_development'" >> $ENV_FILENAME
echo "export MONGODB_PASSWORD='$(vault read secret/kdux/scp/development/$BROAD_USER/mongo/user | grep password | awk '{ print $2 }' )'" >> $ENV_FILENAME
echo "export DATABASE_HOST='$(vault read secret/kdux/scp/development/$BROAD_USER/mongo/hostname | grep ip | awk '{ $2=substr($2,2,length($2)-2); print $2 }' )'" >> $ENV_FILENAME
echo "export BYPASS_MONGO_WRITES='yes' " >> $ENV_FILENAME
echo "export BARD_HOST_URL='https://terra-bard-dev.appspot.com'" >> $ENV_FILENAME
echo "Credentials exported... to load, please run 'source ./$ENV_FILENAME' in your shell"
```

Reply:
After chatting, using the script directly with `source` is a better idea.  Here's a version of the script that can work with `source ./setup_mongo_dev.sh`:

```
#! /bin/bash

VAULT_TOKEN_PATH="$1"
if [[ -z "$VAULT_TOKEN_PATH" ]]
then
  echo "You must provide a path to a Github token to proceed"
  exit 1
fi
vault login -method=github token=$(cat $VAULT_TOKEN_PATH)
if [[ $? -ne 0 ]]
then
  echo "Unable to authenticate into vault"
  exit 1
fi

export BROAD_USER=$(whoami)
export MONGODB_USERNAME='single_cell'
export DATABASE_NAME='single_cell_portal_development'
export MONGODB_PASSWORD=$(vault read secret/kdux/scp/development/$BROAD_USER/mongo/user | grep password | awk '{ print $2 }' )
export DATABASE_HOST=$(vault read secret/kdux/scp/development/$BROAD_USER/mongo/hostname | grep ip | awk '{ $2=substr($2,2,length($2)-2); print $2 }' )
export BYPASS_MONGO_WRITES='yes' 
export BARD_HOST_URL='https://terra-bard-dev.appspot.com'
```

Reply:
4230c8a Thanks for pointing out the hidden assumptions in the setup script, Jon. I've tried to describe the prereqs in the comments and added the VAULT_TOKEN_PATH check as you suggested. Instead of writing a .env file, I find it simpler to source the file itself so that's how I suggest using the script:
`source setup_mongo_dev.sh <path to github token file>`